tools/unicode_table_generator.d

```diff
@@ -0,0 +1,1303 @@
+ushort[dchar] toUpperSimpleIndex, toUpperIndex;
+uint[] toTitleTab;
+ushort toTitleTabSimpleLen; //ditto for Title
+ushort[dchar] toTitleSimpleIndex, toTitleIndex;
+
+mixin(mixedCCEntry);
+
+//case folding mapping
+SimpleCaseEntry[] simpleTable;
+FullCaseEntry[] fullTable;
+
+///canonical combining class
+CodepointSet[256] combiningClass;
+//same but packaged per dchar
+ubyte[dchar] combiningMapping;
+
+//unrolled decompositions
+dstring[dchar] canonDecomp;
+dstring[dchar] compatDecomp;
+
+//canonical composition tables
+dchar[] canonicalyComposableLeft;
+dchar[] canonicalyComposableRight;
+
+
+//canonical composition exclusions
+CodepointSet compExclusions;
+
+//property names to discard
+string[] blacklist = [];
+
+enum mixedCCEntry = `
+struct SimpleCaseEntry
+{
+    uint ch;
+    ubyte n, bucket;// n - number in bucket
+
+pure nothrow @nogc:
+
+    @property ubyte size() const
+    {
+        return bucket & 0x3F;
+    }
+    @property auto isLower() const
+    {
+        return bucket & 0x40;
+    }
+    @property auto isUpper() const
+    {
+        return bucket & 0x80;
+    }
+}
+
+struct FullCaseEntry
+{
+    dchar[3] seq;
+    ubyte n, size;// n number in batch, size - size of batch
+    ubyte entry_len;
+
+    @property auto value() const @trusted pure nothrow @nogc return
+    {
+        return seq[0 .. entry_len];
+    }
+}
+
+struct CompEntry
+{
+    dchar rhs, composed;
+}
+
+struct UnicodeProperty
+{
+    string name;
+    ubyte[] compressed;
+}
+
+struct TrieEntry(T...)
+{
+    size_t[] offsets;
+    size_t[] sizes;
+    size_t[] data;
+}
+
+`;
+
+auto fullCaseEntry(dstring value, ubyte num, ubyte batch_size)
+{
+    dchar[3] val;
+    val[0 .. value.length] = value[];
+    return FullCaseEntry(val, num, batch_size, cast(ubyte) value.length);
+}
+
+enum {
+    UnicodeDatabaseDirectory = "ucd-15/",
+    caseFoldingSrc = UnicodeDatabaseDirectory ~ "CaseFolding.txt",
+    blocksSrc = UnicodeDatabaseDirectory ~ "Blocks.txt",
+    propListSrc = UnicodeDatabaseDirectory ~ "PropList.txt",
+    propertyValueAliases = UnicodeDatabaseDirectory ~ "PropertyValueAliases.txt",
+    corePropSrc = UnicodeDatabaseDirectory ~ "DerivedCoreProperties.txt",
+    normalizationPropSrc = UnicodeDatabaseDirectory ~ "DerivedNormalizationProps.txt",
+    scriptsSrc = UnicodeDatabaseDirectory ~ "Scripts.txt",
+    hangulSyllableSrc = UnicodeDatabaseDirectory ~ "HangulSyllableType.txt",
+    unicodeDataSrc = UnicodeDatabaseDirectory ~ "UnicodeData.txt",
+    compositionExclusionsSrc = UnicodeDatabaseDirectory ~ "CompositionExclusions.txt",
+    specialCasingSrc = UnicodeDatabaseDirectory ~ "SpecialCasing.txt"
+}
+
+enum HeaderComment = `//Written in the D programming language
+/**
+ * License: $(HTTP boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ *
+ * Authors: Dmitry Olshansky
+ *
+ */
+// Automatically generated from Unicode Character Database files, DO NOT EDIT
+// https://github.com/dlang/phobos/blob/master/tools/unicode_table_generator.d
```

Comment:
```suggestion
// !!! DO NOT EDIT !!!
// This module is automatically generated from Unicode Character Database files
// See: https://github.com/dlang/phobos/blob/master/tools/unicode_table_generator.d
```

Reply:
Considering people have edited these files in the past?

Sounds like a plan! lol